test/test_references.jl

```diff
@@ -0,0 +1,33 @@
+using Base.Test
+using Mimi
+
+@defcomp Foo begin
+	input = Parameter()
+	intermed = Variable(index=[time])
+end
+
+function timestep(c::Foo, tt::Int)
```

Comment:
Try to remove the type annotation on ``tt`` here, i.e. change this line to ``function timestep(c::Foo, tt)``. I think what is happening is that ``Int`` defaults to ``Int64`` on x64 platforms, but to ``Int32`` on x86 systems. But Mimi calls ``timestep`` always with a ``Int64`` for the second parameter. So when you declare with ``Int``, multi dispatch will fall back to the generic timestep on x86 systems, and your timestep function is never called.